apps/srt-live-transmit.cpp

```diff
@@ -537,14 +539,19 @@ int main(int argc, char** argv)
                     }
                     break;
                 case UriParser::FILE:
-                    if (srt_epoll_add_ssock(pollid,
-                        src->GetSysSocket(), &events))
                     {
-                        cerr << "Failed to add FILE source to poll, "
-                            << src->GetSysSocket() << endl;
-                        return 1;
+                        int con = src->GetSysSocket();
+                        // try to make the standard input non blocking
+                        srcMayBlock = fcntl(con, F_SETFL, fcntl(con, F_GETFL) | O_NONBLOCK) < 0;
```

Comment:
This functionality is better to be moved inside the file source class to maintain the abstraction layer from the system-level function calls.